pkg/rollout/revision.go

```diff
@@ -0,0 +1,57 @@
+package rollout
+
+import (
+	"google.golang.org/api/run/v1"
+)
+
+// Annotations name for information related to the rollout.
+const (
+	StableRevisionAnnotation    = "rollout.cloud.run/stableRevision"
+	CandidateRevisionAnnotation = "rollout.cloud.run/candidateRevision"
+)
+
+// StableRevision returns the stable revision of the Cloud Run service.
+func StableRevision(svc *run.Service) string {
+	stableRevision, ok := svc.Metadata.Annotations[StableRevisionAnnotation]
```

Comment:
detection mechanism should look at `traffic.tag` before annotation.

let's see if we can make annotation more of an "info" to the user